### PR TITLE
Migration Contracts Changes

### PR DESCRIPTION
**Pull request by remedcu**
This PR adds the new Safe Migration Contracts to the deployments of `v1.4.1`. Corresponding changes have been made to the automated tests and reviews as well.

### CODE REVIEW

File: src/__tests__/assets.test.ts [hunk 1]
```diff
@@ -97,8 +97,10 @@ describe('assets/', () => {
       }
 
       describe('networkAddresses', () => {
-        it('should contain the same networks in all files', async () => {
-          const files = versionFiles(version);
+        it('should contain the same networks in all files (exception for migration contracts)', async () => {
```
Comment (nlordell) [inline src/__tests__/assets.test.ts:100]: nit: We should not ignore these files, but verify that they are a strict subset of the other files.

In particular, you should not be able to have chain FOO in `safe_migration.json` if it is not in `safe.json`.

File: src/__tests__/assets.test.ts [hunk 2]
```diff
@@ -97,8 +97,10 @@ describe('assets/', () => {
...
+          const filesWithMigration = versionFiles(version);
+          const migrationContracts = ['safe_migration.json', 'safe_to_l2_migration.json', 'safe_to_l2_setup.json'];
```
[the remark this answers is not in the document]
Comment (remedcu) [reply src/__tests__/assets.test.ts]: Done 👍🏾

File: src/__tests__/assets.test.ts (hunk 1 above)
Comment (remedcu) [reply src/__tests__/assets.test.ts]: Wrote a new test for this case 👍🏾

File: scripts/review/diff.ts [hunk 3]
```diff
@@ -32,6 +32,41 @@ function hasChangeTypes(changes: parseDiff.Chunk['changes'], types: string[]): b
   return changes.length === types.length && changes.every(({ type }, i) => type === types[i]);
 }
 
+function highestChainID(changes: parseDiff.Chunk['changes']) {
+  assert(hasChangeTypes(changes, ['del', 'add', 'add']));
+  assert(changes[0].content.replace(/^-(.*)/, '+$1,') === changes[1].content);
+}
```
Comment (mmv08) [inline scripts/review/diff.ts:38]: nit: it's unclear to me what this function does, I'm looking at the name and at the function body and can't find a correlation

Comment (remedcu) [reply scripts/review/diff.ts:58]: I have added some comments for more details.

Comment (mmv08) [reply scripts/review/diff.ts:58]: Thanks, it's much better now. I think if the function combines assertions, it'd better be named `assertHighestChainIDChangesValidity` or similar

Comment (remedcu) [reply scripts/review/diff.ts]: Done @mmv08 👍🏾

File: scripts/review/diff.ts [hunk 4]
```diff
@@ -10,6 +10,10 @@ type Options = {
 
 type Deployments = { [chainId: number]: string | string[] };
 
+/**
+ *
```
Comment (nlordell) [inline scripts/review/diff.ts:14]:
```suggestion
```

File: scripts/review/diff.ts [hunk 5]
```diff
@@ -28,10 +32,70 @@ function parseOptions(): Options {
   };
 }
 
+/**
+ *
+ * @param changes The changes in the diff patch.
+ * @param types The types of changes to check.
+ *
+ * This function checks if the changes in the diff patch are of the given types.
```
Comment (nlordell) [inline scripts/review/diff.ts:40]:
```suggestion
 * Check if the changes in the diff patch are of the given types.
 *
 * @param changes The changes in the diff patch.
 * @param types The types of changes to check.
```

This is typically how JSdoc is formatted.

Comment (nlordell) [reply scripts/review/diff.ts:40]: Same for other JSDoc comments.

**Review by nlordell — COMMENTED** [review]
Some more changes, and CI looks to be failing.

File: scripts/review/diff.ts (hunk 5 above)
Comment (remedcu) [reply scripts/review/diff.ts]: Updated 👍🏾

**Comment by remedcu** [comment]
> Some more changes, and CI looks to be failing.

CI will fail because we are changing the asset but not adding a new chain ID or deployments.

File: src/__tests__/assets.test.ts [hunk 6]
```diff
@@ -115,6 +118,41 @@ describe('assets/', () => {
           }
         });
 
+        it('should contain the migration contracts networks in all other files', async () => {
```
Comment (nlordell) [inline src/__tests__/assets.test.ts:121]: IMO this can be part of the above test, most of the counting is duplicated anyway.

Comment (remedcu) [reply src/__tests__/assets.test.ts:121]: I initially thought so, but as the intention of the tests were different, thought to have it separated (one to make sure all the deployments are present in all contracts except the migration one, while the other test makes sure the migration test deployments are also present in other contracts.)

File: scripts/review/diff.ts [hunk 7]
```diff
@@ -28,10 +31,67 @@ function parseOptions(): Options {
   };
 }
 
+/**
+ * Check if the changes in the diff patch are of the given types.
+ *
+ * @param changes The changes in the diff patch.
+ * @param types The types of changes to check.
+ */
```
Comment (mmv08) [inline scripts/review/diff.ts:39]: Nitpick: the JSDoc documents function parameters but doesn't document the return value. The same applies to other JSDocs in the file

**Review by mmv08 — APPROVED** [review]

**Pull request merged**